reconcile/templating/lib/rendering.py

```diff
@@ -43,11 +43,17 @@ def __init__(
         template: Template,
         data: TemplateData,
         secret_reader: SecretReaderBase | None = None,
+        trim_blocks: bool = False,
+        lstrip_blocks: bool = False,
+        keep_trailing_newline: bool = False,
```

Comment:
since these 3 params always bundle together, should we extract a class like `TemplateRenderOptionsV1` but not optional fields?

Reply:
good idea. Applied in [fc5c62f](https://github.com/app-sre/qontract-reconcile/pull/4471/commits/fc5c62fed3dcf42244673e740cad381e6324d395)

Reply:
`TemplateRenderOptions` is frozen, how about create a builder method and reuse it like

```python
    @classmethod
    def create(cls, options: TemplateRenderOptionsV1):
        return cls(
            trim_blocks=options.trim_blocks or False,
            lstrip_blocks=options.lstrip_blocks or False,
            keep_trailing_newline=options.keep_trailing_newline or False,
        )
```

and simplify usage of `env` using `template_render_options.dict()`?

Reply:
Also a good idea. Take a look at the last 2 commits 🙇

Reply:
now I really miss object mapper to auto handle nullable fields convert in other langs like

```
template.template_render_options?.to_model() ?? TemplateRenderOptions()
```

instead of 

```python
TemplateRenderOptions.create(
                trim_blocks=template.template_render_options.trim_blocks
                if template.template_render_options
                else None,
                lstrip_blocks=template.template_render_options.lstrip_blocks
                if template.template_render_options
                else None,
                keep_trailing_newline=template.template_render_options.keep_trailing_newline
                if template.template_render_options
                else None,
            )
```